### PR TITLE
Fix handling of depth+stencil formats in shader validation

### LINKED ISSUE #1491: Reading D24_S8 as input attachment with stencil triggers incorrect validation.
If the stencil aspect of a D24_S8 image is read as UINT (with a VkImageView using only stencil aspect), incorrect validation occurs.

```
            } else if (get_format_type(rpci->pAttachments[index].format) != get_fundamental_type(module, use.second.type_id)) {
                if (log_msg(report_data, VK_DEBUG_REPORT_ERROR_BIT_EXT, VkDebugReportObjectTypeEXT(0), 0, __LINE__,
                            SHADER_CHECKER_INPUT_ATTACHMENT_TYPE_MISMATCH, "SC",
                            "Subpass input attachment %u format of %s does not match type used in shader `%s`", use.first,
                            string_VkFormat(rpci->pAttachments[index].format), describe_type(module, use.second.type_id).c_str())) {
                    pass = false;
                }
            }
```

get_format_type() thinks D24_S8 is FLOAT, but it could be both in a shader.

### PR DESCRIPTION
**Pull request by chrisforbes**
Fixes #1491.

### CODE REVIEW

File: layers/core_validation.cpp [hunk 1]
```diff
@@ -1772,7 +1775,7 @@ static bool validate_vi_against_vs_inputs(debug_report_data *report_data, VkPipe
             unsigned input_type = get_fundamental_type(vs, it_b->second.type_id);
 
             // Type checking
-            if (attrib_type != FORMAT_TYPE_UNDEFINED && input_type != FORMAT_TYPE_UNDEFINED && attrib_type != input_type) {
+            if (!(attrib_type & input_type)) {
```
Comment (tobine) [inline layers/core_validation.cpp]: Previously if either type was UNDEFINED then  we'd skip the error, but now if either (or both) is UNDEFINED we flag the error. Just want to make sure that's intended. Same comment below.

Comment (chrisforbes) [inline layers/core_validation.cpp]: I'm pretty sure that was just an accident of history.

**Comment by mark-lunarg** [comment]
Hey @chrisforbes, is this one ready to go?

**Comment by chrisforbes** [comment]
@mark-lunarg should be good to go.

**Pull request merged**